### PR TITLE
Add docs, update to async/await compatible version of VaporTypedRoutes and add async/await tests

### PR DESCRIPTION
**Pull request by tetraoxygen**
Docs and tests for async/await are currently in this PR, but the async/await version is currently mapped to my fork, pending this PR to VaporTypedRoutes: https://github.com/mattpolzin/VaporTypedRoutes/pull/3

### CODE REVIEW

**Comment by tetraoxygen** [comment]
This should also solve #12.

**Comment by tetraoxygen** [comment]
This should be ready to review — aside from the docs stuff, there actually weren't many changes to this repo in this PR; async/await was basically handled because of mattpolzin/VaporTypedRoutes.

File: Sources/VaporOpenAPI/QueryParam+OpenAPI.swift [hunk 1]
```diff
@@ -7,7 +7,9 @@
 
 import OpenAPIKit
 
+/// A protocol for OpenAPI Arrays.
```
Comment (mattpolzin) [inline Sources/VaporOpenAPI/QueryParam+OpenAPI.swift]:
```suggestion
/// A protocol meant only for internal use.
/// Allows generic constraints and runtime casts on arrays
/// even though concrete arrays have an associated type.
```

**Review by mattpolzin — APPROVED** [review]
Looks great. I nit-picked one of the code docs, but otherwise I am ready to merge.

**Comment by tetraoxygen** [comment]
Awesome, thank you!

**Pull request merged**